processing/src/main/java/org/apache/druid/timeline/DataSegment.java

```diff
@@ -196,6 +197,37 @@ public DataSegment(
       @JsonProperty("size") long size,
       @JacksonInject PruneSpecsHolder pruneSpecsHolder
   )
+  {
+    this(
```

Comment:
Please add a comment here mentioning that we take `interval` input as String so that we can deserialize it optimally.